src/test/java/org/jenkinsci/plugins/durabletask/executors/ContinuedTaskTest.java

```diff
@@ -51,15 +54,14 @@
         final AtomicInteger cntB = new AtomicInteger();
         p.getBuildersList().add(new TestBuilder() {
             @Override public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
-                assertEquals(0, cntA.get());
                 assertEquals(1, cntB.get());
                 return true;
             }
         });
         QueueTaskFuture<FreeStyleBuild> b1 = p.scheduleBuild2(0);
         r.jenkins.getQueue().schedule(new TestTask(cntA, false), 0);
-        r.jenkins.getQueue().schedule(new TestTask(cntB, true), 1);
-        // cntB task ought to run first, then b1, then the cntA task
+        r.jenkins.getQueue().schedule(new TestTask(cntB, true), 0);
+        // cntB task ought to run first, then b1 and the cntA task in either order
```

Comment:
If there was some reason why `b1` would have to run before `cntA`, I have forgotten it. Seems like the order should be arbitrary.